src/services/test_results/TestResultsBuilder.ts

```diff
@@ -1,64 +1,144 @@
 import {TestResultTestField} from '../../dto/test_results/TestResultTestField';
-import {TestResultTestRunElement} from '../../dto/test_results/TestResultTestRunElement';
+import {TestResultUnitTestRunElement} from '../../dto/test_results/TestResultUnitTestRunElement';
 import {TestResultError} from '../../dto/test_results/TestResultError';
 import {WebApi} from 'azure-devops-node-api';
 import * as ba from 'azure-devops-node-api/BuildApi';
 import * as ta from 'azure-devops-node-api/TestApi';
 import {TestCaseResult} from "azure-devops-node-api/interfaces/TestInterfaces";
 import {LogUtils} from "../../LogUtils";
+import {TestResultGherkinTestRunElement} from "../../dto/test_results/TestResultGherkinTestRunElement";
+import {TestResultGherkinTestRunAttributes} from "../../dto/test_results/TestResultGherkinTestRunAttributes";
+import {GherkinTestResultData} from "../../dto/test_results/GherkinTestResultData";
 
 let convert = require('xml-js');
 let xmlescape = require('xml-escape');
+const globby = require('globby');
+const {readFileSync} = require('fs-extra');
 
 export class TestResultsBuilder {
+    private static readonly CUCUMBER_TEST_TYPE = "Cucumber";
 
-    public static buildTestResult(testResults: any, server_id: string, job_id: string, logger: LogUtils): TestResult {
+    public static buildUnitTestResult(testResults: any, server_id: string, job_id: string, logger: LogUtils): TestResult {
         if (!testResults || !testResults.length) {
-            logger.info('No test results were retrieved/found');
+            logger.info('No unit test results were retrieved/found');
             return null;
         }
-        let testResultTestRuns = this.buildTestResultTestRun(testResults, logger);
+        let testResultTestRuns = this.buildUnitTestResultTestRun(testResults, logger);
         let testResultTestFields = this.buildTestResultTestFields(testResults[0].automatedTestType);
         let testResultBuild = this.buildTestResultBuild(server_id, job_id, testResults[0].build.id);
         return new TestResult(testResultBuild, testResultTestFields, testResultTestRuns);
     }
 
-    public static getTestResultXml(testResults: any, server_id: string, job_id: string, logger: LogUtils): any {
-        let result: TestResult = TestResultsBuilder.buildTestResult(testResults, server_id, job_id, logger);
+    public static buildGherkinTestResult(testResults: GherkinTestResultData[], server_id: string, job_id: string, build_id: string, logger: LogUtils): TestResult {
+        if (!testResults || !testResults.length) {
+            logger.info('No gherkin test results were retrieved/found');
+            return null;
+        }
+        let testResultTestRuns = this.buildGherkinTestResultTestRun(testResults, logger);
+        let testResultTestFields = this.buildTestResultTestFields(this.CUCUMBER_TEST_TYPE);
+        let testResultBuild = this.buildTestResultBuild(server_id, job_id, build_id);
```

Comment:
buildTestResultBuild, I don't like it. I don't understand what this is doing. I know it existed in the same way, but think of a better name.